rest/route/host_agent.go

```diff
@@ -525,6 +525,26 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 				"host_id":   currentHost.Id,
 			}))
 
+			if nextTask.IsStuckTask() {
+				if err := model.EndAndResetSystemFailedTask(ctx, env.Settings(), nextTask, evergreen.TaskDescriptionStranded); err != nil {
+					return nil, false, errors.Wrap(err, "ending and resetting system failed task")
+				}
+				// The agent requesting the same task over and over again is a sign that the agent may have hit a panic
```

Comment:
I don't think we need to be so specific with what we guess is the cause. The agent could be doing this for other reasons, like having a non-panic-related bug, or something unusual going on in the host that shuts down the agent. I think it would suffice to say something short and simple to the effect of (but maybe not with these exact words) "the agent is expected to run the task once it's been assigned it. If it's asking for to run the same next task over and over after it's been assigned that task before, there's an issue going on".